lib/src/rules/eol_at_end_of_file.dart

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../analyzer.dart';
+
+const _desc = r'Put a single newline at end of file.';
+
+const _details = r'''
+
+**DO** put a single newline at the end of non-empty files.
+
+**BAD:**
+```dart
+a {
+}
+```
+
+**GOOD:**
+```dart
+b {
+}
+    <-- newline
+''';
+
+class EolLast extends LintRule implements NodeLintRule {
```

Comment:
We usually name the class the same as the rule, but I'm fine for that to get cleaned up in a later PR, either from you or someone else.

Reply:
Thanks, I noted it, and it was committed :)